datastore/key.go

```diff
@@ -254,6 +254,9 @@ func DecodeKey(encoded string) (*Key, error) {
 
 	ref := new(pb.Reference)
 	if err := proto.Unmarshal(b, ref); err != nil {
+		if k := decodeCloudKey(encoded); k != nil {
```

Comment:
Might be worth a comment explaining why this is done this way.

Reply:
Done